Tasks/AzureFunctionOnKubernetesV0/src/utils/commandHelper.ts

```diff
@@ -0,0 +1,112 @@
+import * as tl from "azure-pipelines-task-lib/task";
+import * as tr from "azure-pipelines-task-lib/toolrunner";
+import * as path from "path";
+import { DockerConnection } from "../dockerConnection";
+
+export class CommandHelper {
+    private funcPath: string;
+    public kubectlPath: string;
+
+    constructor() {
+        // Find the path of func and kubectl. This will throw if any of them is not present
+        // and hence the task will fail.
+        this.funcPath = tl.which('func', true);
+        this.kubectlPath = tl.which('kubectl', true);
+    }
+
+    public checkForErrors(execResults: tr.IExecSyncResult[], warnIfError?: boolean) {
+        if (execResults.length !== 0) {
+            let stderr = '';
+            execResults.forEach(result => {
+                if (result.stderr) {
+                    if (result.code !== 0) {
+                        stderr += result.stderr + '\n';
+                    } else {
+                        tl.warning(result.stderr);
+                    }
+                }
+            });
+            if (stderr.length > 0) {
+                if (!!warnIfError) {
+                    tl.warning(stderr.trim());
+                } else {
+                    throw new Error(stderr.trim());
+                }
+            }
+        }
+    }
+
+    public execCommand(command: tr.ToolRunner, options?: tr.IExecOptions, warnIfError?: boolean) {
+        const result: tr.IExecSyncResult = command.execSync(options);
+        this.checkForErrors([result], warnIfError);
+        return result;
+    }
+    
+    public getFuncDeployCommand(dockerConnection: DockerConnection, secretName: string, appName: string, namespace: string, imageName: string, registry: string, pullSecretName: string, args: string): tr.ToolRunner {
```

Comment:
This function can be moved to a `func-object-utility` class in kubernetes common to keep the command logic simple

Reply:
Done. Created funckubernetesutility.ts